core/Rename.cpp

```diff
@@ -142,10 +258,27 @@ namespace olympia
                     const auto rf  = determineRegisterFile(src);
                     const auto num = src.field_value;
                     auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
-                    bitmask.set(num);
+                    uint32_t prf;
+
+                    bool no_prf = false;
+                    if(!map_table_[rf][num].second){
+                        prf = num;
```

Comment:
Does this line indicate PRF#1, PRF#2, ... PRF#31 are reserved for ARF x1, x2, ... x31?

Reply:
Sorry this was older code, Knute and I have discussed that the ARF register should be mapped to the PRF (x0 - x31 mapped to PRF0 - PRF31), so this code isn't applicable anymore. When I wrote this i was under the assumption that once an instruction retired and no references and no mappings, then we would point to the ARF instead, but we're now assuming there is always a valid ARF->PRF mapping.